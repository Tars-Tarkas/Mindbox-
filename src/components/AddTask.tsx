import React, { useContext } from "react";
import type { ToDoContextType } from "../types/todo";
import type { FormProps } from "antd";
import { ToDoContext } from "../context/TodoContext";
import { Form, Input } from "antd";

type FieldType = {
  text?: string;
};

export default function AddTask() {
  const [form] = Form.useForm();
  const { addTodo } = useContext(ToDoContext) as ToDoContextType;

  const onFinish: FormProps<FieldType>["onFinish"] = (values) => {
    addTodo(values.text!);
    form.resetFields();
    form.setFieldValue("text", "");
  };

  return (
    <Form onFinish={onFinish} name="addToDo" autoComplete="off" form={form}>
      <Form.Item<FieldType> name="text">
        <Input autoFocus />
      </Form.Item>
    </Form>
  );
}
