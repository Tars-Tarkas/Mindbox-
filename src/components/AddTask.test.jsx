import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import TodoProvider from "../context/TodoContext";
import AddTask from "./AddTask";

describe("AddTask component", () => {
  const onFinish = jest.fn();
  it("AddTask render", () => {
    render(
      <TodoProvider>
        <AddTask />
      </TodoProvider>
    );
  });
  it("does not submit an empty form", () => {
    render(
      <TodoProvider>
        <AddTask />
      </TodoProvider>
    );
    expect(onFinish).not.toHaveBeenCalled();
  });
  it("AddTask snapshot", () => {
    const addTask = render(
      <TodoProvider>
        <AddTask />
      </TodoProvider>
    );
    expect(addTask).toMatchSnapshot();
  });
});
