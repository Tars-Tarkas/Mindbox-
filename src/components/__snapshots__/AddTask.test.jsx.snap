// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask component AddTask snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
        class="ant-form ant-form-horizontal css-dev-only-do-not-override-1hpnbz2"
        id="addToDo"
      >
        <div
          class="ant-form-item css-dev-only-do-not-override-1hpnbz2"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
          >
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input css-dev-only-do-not-override-1hpnbz2 ant-input-outlined"
                    id="addToDo_text"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-1hpnbz2"
      id="addToDo"
    >
      <div
        class="ant-form-item css-dev-only-do-not-override-1hpnbz2"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-1hpnbz2"
        >
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-1hpnbz2"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input css-dev-only-do-not-override-1hpnbz2 ant-input-outlined"
                  id="addToDo_text"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
